File: ios/Classes/ProtectAppPlugin.swift
import Foundation
import Flutter
import UIKit
import Foundation
import Network

public class ProtectAppPlugin: NSObject, FlutterPlugin {
    private static var screenCaptureDetector: ScreenCaptureDetector?
    
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "protect_app", binaryMessenger: registrar.messenger())
        let instance = ProtectAppPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
        
        // Register event channel for screen capture detection
        let eventChannel = FlutterEventChannel(name: "protect_app/screen_capture", binaryMessenger: registrar.messenger())
        screenCaptureDetector = ScreenCaptureDetector()
        eventChannel.setStreamHandler(screenCaptureDetector)
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "isDeviceUseVPN":
            result(VPNManager.isVpnActive())
        case "dataOfCurrentProxy":
            result(ProxyManager.getProxyData())
        case "checkIsTheDeveloperModeOn":
            result(DeveloperModeManager.isDeveloperModeEnabled())
        case "isItRealDevice":
            result(DeveloperModeManager.isItRealDevice())
        case "isUseJailBrokenOrRoot":
            result(DeveloperModeManager.isJailbroken())
        case "isRunningInTestFlight":
            result(DeveloperModeManager.isRunningInTestFlight())
        case "turnOffScreenshots":
            if let window = UIApplication.shared.windows.first {
                TurnOffScreenshots.turnOff(on: window)
                result(nil)
            } else {
                result(FlutterError(code: "UNAVAILABLE", message: "Window is not available", details: nil))
            }
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}

File: android/src/main/kotlin/com/app/protect_app/ProtectAppPlugin.kt
package com.app.protect_app

import android.app.Activity
import android.content.Context
import android.view.WindowManager
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler

class ProtectAppPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {

  private lateinit var channel: MethodChannel
  private lateinit var eventChannel: EventChannel
  private lateinit var context: Context
  private var activity: Activity? = null
  private var screenCaptureDetector: ScreenCaptureDetector? = null
  
  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    context = flutterPluginBinding.applicationContext
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "protect_app")
    channel.setMethodCallHandler(this)
    
    // Set up event channel for screenshot detection
    eventChannel = EventChannel(flutterPluginBinding.binaryMessenger, "protect_app/screen_capture")
    screenCaptureDetector = ScreenCaptureDetector(context, activity)
    eventChannel.setStreamHandler(screenCaptureDetector)
  }

  override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
    when (call.method) {
      "isDeviceUseVPN" -> result.success(VPNManager.isDeviceUseVPN(context))
      "dataOfCurrentProxy" -> result.success(ProxyManager.dataOfCurrentProxy(context))
      "checkIsTheDeveloperModeOn" -> result.success(DeveloperModeManager.isDeveloperModeEnabled())
      "turnOffScreenshots" -> {
        activity?.window?.setFlags(WindowManager.LayoutParams.FLAG_SECURE, WindowManager.LayoutParams.FLAG_SECURE)
        result.success(null)
      }
      "isItRealDevice" -> result.success(DeveloperModeManager.isItRealDevice())
      "isUseJailBrokenOrRoot" -> result.success(DeveloperModeManager.isUseJailBrokenOrRoot())
      "isRunningInTestFlight" -> result.success(false) // Only for iOS
      else -> result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
    eventChannel.setStreamHandler(null)
    screenCaptureDetector?.dispose()
    screenCaptureDetector = null
  }

  override fun onAttachedToActivity(binding: ActivityPluginBinding) {
    activity = binding.activity
    screenCaptureDetector?.updateActivity(activity)
  }

  override fun onDetachedFromActivityForConfigChanges() {
    activity = null
    screenCaptureDetector?.updateActivity(null)
  }

  override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
    activity = binding.activity
    screenCaptureDetector?.updateActivity(activity)
  }

  override fun onDetachedFromActivity() {
    activity = null
    screenCaptureDetector?.updateActivity(null)
  }
}
